Docs-978 fully branched PGD 5.6 (#6017)
EnterpriseDB/docs: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx

Each subscriber-only group uses that group leader to replicate changes to other subscriber-only nodes in its group. The group leader acts as a replication proxy for incoming changes.

The write leader nodes in other non-subscriber-only groups replicate changes to the group leader of the subscriber-only group. Other nodes only replicate within their data groups, both within their own group and in other data groups.

## Subscriber-only group leaders

With PGD 5.6 and later, each subscriber-only group gets assigned a group leader of its own. 
This is because subscriber-only groups don't have a group Raft consensus mechanism of their own. 
Instead, the cluster’s top level group uses its Raft consensus mechanism to handle selection of each subscriber-only group’s group leader.
This group leader selection is on by default on PGD 5.6, regardless of the topology optimization settings.

Group leaders in subscriber-only groups are regularly tested for connectivity and if unavailable, the voting nodes of top-level group select a new subscriber-only node from the subscriber-only group to become group leader. 
The new group leader is then selected. 

> With optimized technology turned off, this election has no effect on the replication topology. Without the optimized topology, all nodes replicates changes to all other nodes in the  cluster.

Reviewer: Change: 
Without the optimized topology, all nodes replicates changes to all other nodes in the cluster
To:
Without the optimized topology, all data nodes replicates changes to all other nodes in the cluster

Author: ```suggestion
With optimized technology turned off, this election has no effect on the replication topology. Without the optimized topology, all data nodes replicate changes to all other nodes in the cluster.
```

## Group leaders in the optimized topology

>  With the optimized topology enabled, only the group leader receives changes from other nodes in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group. 

Reviewer: It is not clear which group leader is referred here.

Author: ```suggestion
 With the optimized topology enabled, only the subscriber-only group's group leader receives changes from other nodes in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group. 
```

>  With the optimized topology enabled, only the subscriber-only group's group leader receives changes from other nodes in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group. 

Reviewer: ```suggestion
 With the optimized topology enabled, only the subscriber-only group's group leader receives changes from other data group's write leaders in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group. 
```

The other voting nodes choose the group leader from a subscriber-only group's nodes.
Once selected, the whole cluster becomes aware of the change and any data group's write leaders then replicate data only to this newly selected group leader node. 
Other data nodes in the data groups don't replicate data to the subscriber-only group's nodes.

> This approach avoids the explosion of connections that can happen when there are large numbers of SO nodes and reduces the amount of replication traffic.

Reviewer: In general connectivity between all nodes is required as any node can become a leader. 
Hence should we use wording as "explosion of active connections" or "connections in use". Checking with Vaijayanti too on this.

Author: ```suggestion
This approach avoids the explosion of active connections that can happen when there are large numbers of SO nodes and reduces the amount of replication traffic.
```
